### PR TITLE
ThreadAndName

### PR DESCRIPTION
**Pull request by marceloaguiarr**
Changes:

 * Changed class name
 * Implemented the Initialization-on-demand idiom

### CODE REVIEW

**Comment by MAAARKIN** [comment]
There some way to you make a benchmark to show the better performance over the synchronized?

File: heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/trace/TraceContextHolder.java [hunk 1]
```diff
@@ -42,23 +42,28 @@
...
      /**
-      * Thread safe singleton initializer.
+      * Implementation of the Initialization-on-demand holder idiom.
+      * 
+      * @author Marcelo Rodrigues
+      * @see <a href="http://literatejava.com/jvm/fastest-threadsafe-singleton-jvm/">Fastest Thread-safe Singleton in Java</a>
       * 
-      * @return		{@link TraceContextHolder} instantce
       */
-     public static synchronized TraceContextHolder getInstance() {
```
Comment (thiagonego) [inline heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/trace/TraceContextHolder.java]: @filipegermano what do you think about that? :

Comment (marceloaguiarr) [reply heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/trace/TraceContextHolder.java]: I have created a simple benchmark for three idioms for singleton initialization. Here is a link for the repository: [link](https://github.com/marceloaguiarr/SingletonBenchmark)

In a nutshell, the synchronized keyword used in the method signature means that every single call to this method will have a synchronization overhead. This is great for the first few thread that call the getInstance method, but all others call, when the instance is already alive, shouldn't have to create such overhead.

By using the Initialization on demand idiom we can achieve lazy initialization (the instance is only created when needed) with thread safety guarantied by the JVM class creation. No synchronization overhead is needed and every other call made after the instance is created is as simple as a memory lookup.

Comment (marceloaguiarr) [reply heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/trace/TraceContextHolder.java]: Finally had time to actually run the benchmarks and got the results.

Logs are available in the repository.

**Comment by MAAARKIN** [comment]
Solve the #24

**Pull request merged**